desktop/src/Desktop/ImportExport.hs

```
@@ -0,0 +1,234 @@
+import Desktop.Orphans ()
+import Desktop.Crypto.BIP (BIPStorage(..), bipMetaPrefix, runBIPCryptoT)
+import Frontend.AppCfg (ExportWalletError(..), FileType(FileType_Import), fileTypeExtension)
+import Pact.Server.ApiClient (TransactionLogger (..), CommandLog, commandLogCurrentVersion)
+import Frontend.Crypto.Class (HasCrypto)
+import Frontend.Storage (HasStorage, dumpLocalStorage)
+import Frontend.VersionedStore (StoreFrontend(..), VersionedStorage(..), StorageVersion, VersioningDecodeJsonError(..))
+import qualified Frontend.VersionedStore as FrontendStore
+
+newtype Password = Password { unPassword :: Text } deriving (Eq)
+
+data ImportWalletError
+  = ImportWalletError_PasswordIncorrect
+  | ImportWalletError_NoRootKey
+  | ImportWalletError_NotJson Text
+  | ImportWalletError_UnknownVersion Text StorageVersion
+  | ImportWalletError_DecodeError Text StorageVersion Text
+  | ImportWalletError_InvalidCommandLogDestination
+  | ImportWalletError_CommandLogWriteError
+  deriving (Eq, Show)
+
+bipStorageVersionKey :: Text
+bipStorageVersionKey = "BIPStorage_Version"
+
+bipStorageDataKey :: Text
+bipStorageDataKey = "BIPStorage_Data"
+
+storeFrontendVersionKey :: Text
+storeFrontendVersionKey = "StoreFrontend_Version"
+
+storeFrontendDataKey :: Text
+storeFrontendDataKey = "StoreFrontend_Data"
+
+commandLogDataKey :: Text
+commandLogDataKey = "CommandLogs_Data"
+
+commandLogVersionKey :: Text
+commandLogVersionKey  = "CommandLogs_Version"
+
+hoistParser
+  :: Monad m
+  => Text
+  -> StorageVersion
+  -> (Value -> Parser a)
+  -> Value
+  -> ExceptT ImportWalletError m a
+hoistParser errLabel ver p  =
+  hoistEither
+  . first (ImportWalletError_DecodeError errLabel ver . T.pack)
+  . parseEither p
+
+extractImportVersionField
+  :: (Monad m)
+  => Text
+  -> StorageVersion
+  -> Value
+  -> ExceptT ImportWalletError m StorageVersion
+extractImportVersionField key ver =
+  hoistParser key ver (withObject "ChainWeaverImport" (\o -> o .:? key .!= 0 ))
```

Don't we always put the version there? 
Seems like a missing version would be unexpected, in which case better fail fast.